tests/borrow.rs

```diff
@@ -107,16 +108,17 @@ fn borrows_span_frames() {
 #[test]
 fn borrows_span_threads() {
     Python::with_gil(|py| {
+        use pyo3::PyNativeType;
         let array = PyArray::<f64, _>::zeros(py, (1, 2, 3), false);
 
         let _exclusive = array.readwrite();
 
-        let array = array.to_owned();
+        let array = (*array.as_borrowed()).clone().unbind();
```

Comment:
Is this is the same `to_owned` on which you allowed the deprecation warning above? Why are they handled differently?

Reply:
The one I allowed was a test specific for `to_owned`, this one is not

Reply:
That makes sense. Could you also add a fixme comment here w.r.t. `to_owned().unbind()` when released? (It should apply here, shouldn't it?)

Reply:
We won't need it anymore, once we have `zeros_bound`, which is why I did not put the comment here

Reply:
Ok, so this ties in with https://github.com/PyO3/rust-numpy/pull/415#discussion_r1536611024

Reply:
> I have split off constructors for later PRs to keep the size of this manageable.

In this case, could you immediately convert the GIL ref returned by `zeros` so that this part already has its final form?